test/src/main/java/org/apache/accumulo/test/functional/PermissionsIT.java

```diff
@@ -780,4 +784,54 @@ private void verifyHasNoTablePermissions(AccumuloClient root_client, String user
         throw new IllegalStateException(
             user + " SHOULD NOT have table permission " + p + " for table " + table);
   }
+
+  private void testArbitraryProperty(AccumuloClient c, String tableName, boolean havePerm)
+      throws AccumuloException, TableNotFoundException {
+    // Set variables for the property name to use and the initial value
+    String propertyName = "table.custom.description";
+    String description1 = "Description";
+
+    // Make sure the property name is valid
+    assertTrue(Property.isValidPropertyKey(propertyName));
+    // Set the property to the desired value
+    try {
+      c.tableOperations().setProperty(tableName, propertyName, description1);
+
+      // Loop through properties to make sure the new property is added to the list
+      int count = 0;
+      for (Entry<String,String> property : c.tableOperations().getProperties(tableName)) {
+        if (property.getKey().equals(propertyName) && property.getValue().equals(description1))
+          count++;
+      }
+      assertEquals(count, 1);
+
+      // Set the property as something different
+      String description2 = "set second";
+      c.tableOperations().setProperty(tableName, propertyName, description2);
+
+      // Loop through properties to make sure the new property is added to the list
+      count = 0;
+      for (Entry<String,String> property : c.tableOperations().getProperties(tableName)) {
+        if (property.getKey().equals(propertyName) && property.getValue().equals(description2))
+          count++;
+      }
+      assertEquals(count, 1);
+
+      // Remove the property and make sure there is no longer a value associated with it
+      c.tableOperations().removeProperty(tableName, propertyName);
```

Comment:
Ideally removeProperty would be tested when the user does not have permission.  If the call to setProp fails because the user does not have permission, the exception will prevent this code from executing.  I don't think the exiting test checked this either.  So this should not hold up progress, this change still seems like an overall improvement.

Reply:
Yeah I see what you are saying.  The testMissingSystemPermission() method in this class does test that removeProperty throws a AccumuloSecurityException.  [Here](https://github.com/milleruntime/accumulo/blob/9292f715adaa69748696e115aa2b6246c32218aa/test/src/main/java/org/apache/accumulo/test/functional/PermissionsIT.java#L204)